src/pyk/kcfg/kcfg.py

```diff
@@ -214,6 +217,9 @@ def __init__(self) -> None:
         self._stuck = set()
         self._aliases = {}
         self._lock = RLock()
+        self.cfg_dir = cfg_dir
+        if self.cfg_dir is not None and not self.cfg_dir.exists():
+            self.cfg_dir.mkdir()
```

Comment:
```suggestion
        self.cfg_dir = cfg_dir
        if self.cfg_dir is not None:
            ensure_dir_path(self.cfg_dir)
```

This can be imported as: `from ..utils import ensure_dir_path`